File: Soal1.py
import math

x = 0.3
eksak = math.exp(x)

print("Nilai eksak e^0.3 =", eksak)
print("n\tPendekatan\t\tAbsolute Error\tRelative Error\tRPE")

for n in range(5):
    S = 0

    for i in range(n + 1):
        S += (x ** i) / math.factorial(i)

    AE = abs(eksak - S)
    RE = AE / abs(eksak)
    RPE = RE * 100

    print(f"{n}\t{S:.15f}\t{AE:.15f}\t{RE:.8f}\t{RPE:.8f}%")

File: Soal1.m
% Galat pada perhitungan e^(0.3) menggunakan deret Taylor

format long g

x = 0.3;
eksak = exp(x);

fprintf('Nilai eksak e^0.3 = %.15f\n', eksak);
fprintf('n\t\tPendekatan\t\tAbsolute Error\t\tRelative Error\t\tRPE\n');

for n = 0:4
    S = 0;

    for i = 0:n
        S = S + x^i / factorial(i);
    end

    AE = abs(eksak - S);
    RE = AE / abs(eksak);
    RPE = RE * 100;

    fprintf('%d\t\t%.15f\t%.15f\t%.8f\t%.8f%%\n', ...
        n, S, AE, RE, RPE);
end
